cdap-proto/src/main/java/co/cask/cdap/proto/DatasetInstanceConfiguration.java

```diff
@@ -37,7 +36,7 @@ public String getTypeName() {
   }
 
   public Map<String, String> getProperties() {
-    return properties == null ? ImmutableMap.<String, String>of() : properties;
+    return properties == null ? Collections.unmodifiableMap(Collections.<String, String>emptyMap()) : properties;
```

Comment:
`emptyMap` is already immutable http://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#emptyMap()